gtsam/hybrid/tests/testHybridEstimation.cpp

```diff
@@ -432,6 +432,69 @@ TEST(HybridEstimation, ProbabilityMultifrontal) {
   EXPECT(assert_equal(discrete_seq, hybrid_values.discrete()));
 }
 
+/****************************************************************************/
+/**
+ * Test for correctness via sampling.
+ *
+ * Given the conditional P(x0, m0, x1| z0, z1)
+ * with meaasurements z0, z1, we:
+ * 1. Start with the corresponding Factor Graph `FG`.
+ * 2. Eliminate the factor graph into a Bayes Net `BN`.
+ * 3. Sample from the Bayes Net.
+ * 4. Check that the ratio `BN(x)/FG(x) = constant` for all samples `x`.
+ */
+TEST(HybridEstimation, CorrectnessViaSampling) {
+  HybridNonlinearFactorGraph nfg;
+
+  auto noise_model = noiseModel::Diagonal::Sigmas(Vector1(1.0));
```

Comment:
I thought there was a 1D constructor Sigma

Reply:
Nope, that's only for Isotropic. Which I could use here as well, so I'll update this.